### PR TITLE
solution

### PR DESCRIPTION
**Pull request by ihorandrianov**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1 +1,62 @@
...
+
+
+      file.pipe(gzipCompress);
+      gzipCompress.pipe(write);
+    });
+
+    bb.on('finish', () => {
+      res.statusCode = 200;
+      res.end('OK');
+    });
```
Comment (sergii-nosachenko) [inline src/index.js:36]: Return archived file to client. Or give a link to download compressed file.

File: src/public/index.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  <title>Document</title>
+</head>
+<body>
+  <form
+    action="/"
+    method="POST"
+    enctype="multipart/form-data"
+  >
+    <label for="file"></label>
+    <input type="file" id="file" name="file">
```
Comment (sergii-nosachenko) [inline src/public/index.html:16]: Add ability to select compression type.

File: src/index.js [hunk 3]
```diff
@@ -1 +1,62 @@
...
+  return () => randomFillSync(buf).toString('hex');
+})();
+
+server.on('request', (req, res) => {
+  if (req.method === 'POST') {
+    const bb = busboy({ headers: req.headers });
+
+    bb.on('file', (name, file, info) => {
+      const saveTo = path.join(__dirname, `compressed-${random()}`);
+      const gzipCompress = zlib.createGzip();
```
Comment (sergii-nosachenko) [inline src/index.js:25]: zlib provides a couple of compression methods, not only gzip

File: src/index.js [hunk 4]
```diff
@@ -1 +1,62 @@
...
+
+  return () => randomFillSync(buf).toString('hex');
+})();
+
+server.on('request', (req, res) => {
+  if (req.method === 'POST') {
+    const bb = busboy({ headers: req.headers });
+
+    bb.on('file', (name, file, info) => {
+      const saveTo = path.join(__dirname, `compressed-${random()}`);
```
Comment (sergii-nosachenko) [inline src/index.js:24]: Why not to use original filename with `compressed-`?

File: src/index.js [hunk 5]
```diff
@@ -1 +1,62 @@
...
+const zlib = require('zlib');
+
+const PORT = 8000;
+const server = new http.Server();
+
+const random = (() => {
+  const buf = Buffer.alloc(16);
+
+  return () => randomFillSync(buf).toString('hex');
+})();
```
Comment (sergii-nosachenko) [inline src/index.js:17]: Unnecessary randomizer, do you need it?

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

**Review by Andr1yK — APPROVED** [review]

**Pull request closed without merging**